subsys/net/lib/dhcpv4/dhcpv4_server.c

```diff
@@ -1794,17 +1842,69 @@ int net_dhcpv4_server_foreach_lease(struct net_if *iface,
 	return ret;
 }
 
+/*
+ *	Finds a free address, taking the validator callback into account
+ */
+static struct in_addr net_dhcpv4_server_find_free_address(
+	struct dhcpv4_server_ctx *ctx, struct dhcpv4_client_id *client_id)
+{
+	/* Try to look for next available address */
+	struct net_in_addr address;
+
+	for (unsigned int i = 0; i < CONFIG_NET_DHCPV4_SERVER_ADDR_COUNT; ++i) {
+		address.s_addr = htonl(ntohl(ctx->base_addr.s_addr) + i);
+		bool taken = false;
+
+		/* Skip if address is not assignable by application */
+		if (address_validator_callback &&
+			address_validator_callback(ctx->iface, client_id, &address,
```

Comment:
Please fix the alignment, the if condition contents should align with the opening bracket, not the block content, it's currently hard to read.